src/networking/nc/mod.rs

```diff
@@ -0,0 +1,958 @@
+        let mut portlist = vec!();
+        let lflag = matches.is_present("l");
+        let mut host = String::from("127.0.0.1");
+        let uport:String;
+        let mut interval = None;
+        let mut timeout = None;
+        let s_addr = match matches.value_of("s") {
+            Some(addr) => Some(String::from(addr)),
+            None => None
+        };
+        let uflag = matches.is_present("u");
+        let pflag = matches.is_present("p");
+
+        let zflag = matches.is_present("z");
+        let kflag = matches.is_present("k");
+
+        /* Cruft to make sure options are clean, and used properly. */
+        let positionals: Vec<&str> = if matches.is_present("positionals") {
+            matches.values_of("positionals").unwrap().collect()
+        } else {
+            vec!()
+        };
+
+        let family = if matches.is_present("U") {
+            AF_UNIX
+        } else if matches.is_present("6") {
+            AF_INET6
+        } else if matches.is_present("4") {
+            AF_INET
+        } else {
+            AF_UNSPEC
+        };
+
+        if positionals.len() == 1 {
+            if family == AF_UNIX {
+                host = String::from(positionals[0]);
+                uport = String::new();
+            } else {
+                if !lflag {
+                    return mesaerr_result(msg);
+                }
+                uport = String::from(positionals[0]);
+            }
+        } else if positionals.len() >= 2 {
+            host = String::from(positionals[0]);
+            uport = String::from(positionals[1]);
+        } else {
+            return mesaerr_result(msg);
+        }
+
+        if lflag && s_addr.is_some() {
+            return mesaerr_result("cannot use -s and -l");
+        }
+        if lflag && pflag {
+            return mesaerr_result("cannot use -p and -l");
+        }
+        if lflag && zflag {
+            return mesaerr_result("cannot use -z and -l");
+        }
+        if !lflag && kflag {
+            return mesaerr_result("must use -l with -k");
+        }
+
+        if !uport.is_empty() {
+            portlist = build_ports(&uport)?;
+        }
+
+        if matches.is_present("i") {
+            let sec = matches.value_of("i").unwrap().parse::<u64>().unwrap();
+            interval = Some(Duration::new(sec, 0));
+        }
+
+        if matches.is_present("w") {
+            let sec = matches.value_of("w").unwrap().parse::<u64>().unwrap();
+            timeout = Some(Duration::new(sec, 0));
+        }
+
+        let mut unix_dg_tmp_socket = String::new();
+
+        /* Get name of temporary socket for unix datagram client */
+        if family == AF_UNIX && uflag && !lflag {
+            unix_dg_tmp_socket = if s_addr.is_some() {
+                s_addr.clone().unwrap()
+            } else {
+                let nf = NamedTempFile::new()?;
+                let path = String::from(nf.path().to_str().unwrap());
```

Comment:
Rather than a `String`, this should most like be a `Path` or `PathBuf`.  You might be able to get `Path` to work by giving `NcOptions` a lifetime and *not* taking ownership of `ArgMatches` (_i.e._ use `&mut ArgMatches` instead).

Reply:
Path won't work since nc doesn't only work for unix. For TCP/UDP, the type is not path.

Reply:
Isn’t `unix_dg_tmp_socket` just used for Unix sockets though?  I scanned through the code briefly and didn’t notice anything that indicated otherwise, but let me know if I missed something.

Reply:
Yes `unix_dg_tmp_socket` is only for unix socket. However, after I changed it to Path, I found that I had to immediately convert it to string because the functions consuming it also handle TCP/UDP and cannot take Path. Besides, no Path-related action is performed so there is no benefit to use Path.

Reply:
I know what you mean now. Thank you for the explanation.